tests/test_pipelineTaskTests.py

```diff
@@ -0,0 +1,416 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Unit tests for `lsst.pipe.base.tests`, a library for testing
+PipelineTask subclasses.
+"""
+
+import shutil
+import tempfile
+import unittest
+
+import lsst.utils.tests
+import lsst.pex.config
+import lsst.daf.butler
+import lsst.daf.butler.tests as butlerTests
+
+from lsst.pipe.base import Struct, PipelineTask, PipelineTaskConfig, PipelineTaskConnections, connectionTypes
+from lsst.pipe.base.tests import runQuantum, makeQuantum, validateOutputConnections
+
+
+class VisitConnections(PipelineTaskConnections, dimensions={"instrument", "visit"}):
```

Comment:
I believe we're attempting to phase out `visit` in favor of `expId`. Unless you know otherwise, please do so throughout.

Reply:
I think that is waiting on daf_butler to catch up with the dimensions definitions. visit is allowed when it's really a visit as opposed to a pipeline that's working on individual exposures.

Reply:
I think most of the tasks we're interested in in AP, including anything that acts on a calexp, use visits. Though I haven't seen an explicit specification to that effect...